### ISSUE TITLE
out_file: Proposal for a new fomatter option ("Template Formatter")

### ISSUE THREAD

**Issue by fujimotos** [fluent/fluent-bit#1478]
This is a proposal for a new formatting feature for `out_file`.

**Is your feature request related to a problem? Please describe.**

Right now, we have five formatters for out_file: *ltsv, csv, json, msgpack
and plain*. They are good for many cases, but there are cases that they
cannot handle.

The most notable situation is when we want to output incoming data *as is*.
Suppose we have the following log file.

```
2019-07-01 12:31:20 Service is starting up.
2019-07-01 12:31:21 Waiting when pool is empty...
```

Right now, it is not possible to transfer its content to another file while
preserving the original format. In fact, even `plain` formatter produces
a quite different result (as shown below).

```
{"log":"2019-07-01 12:31:20 Service is starting up."}
{"log":"2019-07-01 12:31:21 Waiting when pool is empty..."}
```

**Describe the solution you'd like**

Add a new formatter option `template` to out_file.

 * It accepts a format string like `"{time} {level} - {message}"`
 * It replaces each placeholder dynamically using the incoming data.

Here is how the new configuration option looks like (the most simplest example).

```ini
# in_tail produces a record like {"log": "this is some log line"}
[INPUT]
  Name     tail
  Path     source.log

# out_file converts the record back to "this is some log line"
[OUTPUT]
  Name     file
  Format   template
  Template {log}
  Path     dest.log
```

This should transfer the content into the destination without breaking
the original format, thus, resolves the problem being described above.

**Comment by fujimotos** [comment]
Note that Fluentd solves the particular problem (discussed above) by providing
a formatter named `single_value`.

https://docs.fluentd.org/formatter/single_value

The `template` formatter is generalization of the concept; Not only can it output
a single field in a record, but also handle an arbitrary output format.

I think it will have a much broader range of applications.

**Comment by fujimotos** [comment]
This is already merged and usable as of v1.3.0.
